packages/web/src/pages/audio-rewards-page/components/modals/ChallengeRewardsModal/ClaimAllRewardsModal.tsx

```diff
@@ -25,23 +39,70 @@ const messages = {
   upcomingRewards: 'Upcoming Rewards',
   claimAudio: (amount: string) => `Claim ${amount} $AUDIO`,
   readyToClaim: 'Ready to claim!',
+  rewardsClaimed: 'All rewards claimed successfully!',
   rewards: 'Rewards',
   audio: '$AUDIO',
   description: 'You can check and claim all your upcoming rewards here.',
   done: 'Done'
 }
 
+const { show: showConfetti } = musicConfettiActions
+const { claimAllChallengeRewards } = audioRewardsPageActions
+const { getClaimStatus, getUndisbursedUserChallenges } =
+  audioRewardsPageSelectors
+
 export const ClaimAllRewardsModal = () => {
+  const dispatch = useDispatch()
+  const { toast } = useContext(ToastContext)
+  const wm = useWithMobileStyle(styles.mobile)
   const [isOpen, setOpen] = useModalState('ClaimAllRewards')
   const [isHCaptchaModalOpen] = useModalState('HCaptcha')
-  const wm = useWithMobileStyle(styles.mobile)
-
-  const { claimableAmount, cooldownChallenges, summary } =
+  const claimStatus = useSelector(getClaimStatus)
+  const undisbursedUserChallenges = useSelector(getUndisbursedUserChallenges)
+  const undisbursedUserChallengesMap = undisbursedUserChallenges.reduce<
```

Comment:
I personally find the reduce here and the reduce in the onClaimRewardClicked a bit confusing. Mind hopping on a huddle to go through it?

Reply:
curious about your thoughts. but this is now removed for other reasons :)